File: Domain/Budget.cs
using Domain.Enums;
using Microsoft.EntityFrameworkCore;

namespace Domain
{
    public class Budget
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public BudgetPeriod Period { get; set; } = BudgetPeriod.Month;
        [Precision(18, 2)]
        public decimal Amount { get; set; }
        public string UserId { get; set; }
        public int CurrencyId { get; set; }

        public virtual User User { get; set; }
        public virtual Currency Currency { get; set; }
        public virtual ICollection<BudgetCategory> Categories { get; set; }
    }
}

File: client-app/src/app/models/Budget.ts
import { Category, CategoryOption } from "./Category";
import { Currency } from "./Currency";
import { BudgetPeriod } from "./enums/BudgetPeriod";

export interface Budget {
    id: number;
    name: string;
    period: BudgetPeriod;
    amount: number;
    currentAmount: number;
    convertedAmount: number;
    currency: Currency;
    categories: Category[];
}

export interface BudgetSummary {
    percentage: number;
    progressValue: number; 
    currentAmount: number; 
    fullAmount: number;
}

export interface BudgetFormValues {
    name: string;
    categories: CategoryOption[];
    period: BudgetPeriod;
    amount: number | null;
}

export interface BudgetDto {
    name: string;
    categoryIds: number[];
    period: BudgetPeriod;
    amount: number | null;
}